resource/policies/base/match_op.h

```diff
@@ -0,0 +1,34 @@
+#ifndef MATCH_OP_H
+#define MATCH_OP_H
+
+
+enum match_op_t { MATCH_UNKNOWN,
+                  MATCH_ALLOCATE,
+                  MATCH_ALLOCATE_W_SATISFIABILITY,
+                  MATCH_ALLOCATE_ORELSE_RESERVE,
+                  MATCH_SATISFIABILITY };
+
+static const char *match_op_to_string (enum match_op_t match_op) {
```

Comment:
No need to specify `enum` type here:
```suggestion
static const char *match_op_to_string (match_op_t match_op) {
```

Reply:
Is this a C++-only header? I thought this was C, where it's required unless there's a `typedef` to alias the enum to the unqualified name.

Reply:
We do usually do that though... Oh I see, it used to be in a C++ file and was moved here.  The equivalent would be to make it:

```c
typedef enum match_op_t { MATCH_UNKNOWN,
                  MATCH_ALLOCATE,
                  MATCH_ALLOCATE_W_SATISFIABILITY,
                  MATCH_ALLOCATE_ORELSE_RESERVE,
                  MATCH_SATISFIABILITY } match_op_t;
```

That way it can be used bare, which is what we do most places.

Reply:
This is super helpful, thanks so much for the suggestion! This is definitely cleaner and more constant with the rest of the code, I just wasn't sure that I could/how to do it before. Went ahead and made those changes, so should be ready now!

Reply:
Yeah, that's a clean solution!